### PR TITLE
Add URL API

### PR DESCRIPTION
**Pull request by enjikaka**
Render: https://ipfs.io/ipfs/QmaL7X5G8ESzhmv45MDJKNg11GDpe1TVVWBtcR2jXTV64u/
Based on:
- https://developer.mozilla.org/en-US/docs/Web/API/URL
- https://developer.microsoft.com/en-us/microsoft-edge/platform/catalog/?q=specName%3Awhatwg-url
- Some own testing in Opera

### CODE REVIEW

File: api/URL.json [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+  "api": {
+    "URL": {
+      "__compat": {
+        "mdn_url": "https://developer.mozilla.org/docs/Web/API/URL",
+        "support": {
+          "chrome": [
+            {
+              "version_added": "2",
+              "alternative_name": "webkitURL",
+              "notes": "This feature is implemented under the non-standard name webkitURL."
```
Comment (Elchi3) [inline api/URL.json]: This note is redundant. Please remove

File: api/URL.json [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+            },
+            {
+              "version_added": "32"
```
Comment (Elchi3) [inline api/URL.json]: Can you swap this around to be reverse chronological, please? 
[
  { version_added: 32  },
  { version_added: 2, alternative_name: webkitURL}
]

File: api/URL.json [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+            "version_added": "12"
+          },
+          "firefox": [
+            {
+              "version_added": "4",
+              "version_removed": "18",
+              "notes": "From Gecko 2 (Firefox 4) to Gecko 18 included, Gecko supported this interface with the non-standard nsIDOMMozURLProperty internal type. As the only to access such an object was through window.URL, in practice, this didn't make any difference."
+            },
+            {
+              "version_added": "19"
```
Comment (Elchi3) [inline api/URL.json]: Reverse chronological please. version 19 is the most relevant version that should appear in the main cell.

File: api/URL.json [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+          ],
+          "ie": {
+            "version_added": false
+          },
+          "opera": [
+            {
+              "version_added": "15",
+              "alternative_name": "webkitURL",
+              "notes": [
+                "This feature is implemented under the non-standard name webkitURL."
```
Comment (Elchi3) [inline api/URL.json]: note is redundant, please remove

File: api/URL.json [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+              ]
+            },
+            {
+              "version_added": "19"
```
Comment (Elchi3) [inline api/URL.json]: This needs to come first

File: api/URL.json [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+            }
+          ],
+          "safari": [
+            {
+              "version_added": "6",
+              "alternative_name": "webkitURL",
+              "notes": [
+                "This feature is implemented under the non-standard name webkitURL."
```
Comment (Elchi3) [inline api/URL.json]: remove

File: api/URL.json [hunk 7]
```diff
@@ -0,0 +1,538 @@
...
+              ]
+            },
+            {
+              "version_added": "7"
```
Comment (Elchi3) [inline api/URL.json]: Put this version first

**Comment by enjikaka** [comment]
@Elchi3 Updated now!

**Comment by Elchi3** [comment]
Wow, nice work! I really like the idea of little helper scripts. Would you mind putting that into a new PR, though? Allows us to have a separate discussion about how we want to add this.

Looks good otherwise except that the constructor is still missing as a sub feature, https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

**Comment by enjikaka** [comment]
> Wow, nice work! I really like the idea of little helper scripts. Would you mind putting that into a new PR, though? Allows us to have a separate discussion about how we want to add this.

I'll make a new PR with that! :)

> as a sub feature

I don't fully understand. Isn't the first __compat the constructor? 🤔

**Comment by Elchi3** [comment]
The first __compat is the URL interface itself (and some interfaces have no constructor).

**Comment by enjikaka** [comment]
@Elchi3 Ok, I'll look over the file again and try to understand what to change and get it sorted! Meanwhile; here's a PR for the helper script: https://github.com/mdn/browser-compat-data/pull/1314

**Comment by Elchi3** [comment]
Thanks! 
Have a look at other APIs, like https://github.com/mdn/browser-compat-data/blob/master/api/AnalyserNode.json which also has a constructor (api.AnalyserNode.AnalyserNode).

**Comment by enjikaka** [comment]
@Elchi3 Added constructor now!

**Pull request merged**